js/storage.js

```diff
@@ -1268,78 +1260,64 @@ Storage.exportTeam = function (team) {
 		if (typeof curSet.happiness === 'number' && curSet.happiness !== 255 && !isNaN(curSet.happiness)) {
 			text += 'Happiness: ' + curSet.happiness + "  \n";
 		}
-		var first = true;
+
+		var firstEV = true;
 		if (curSet.evs) {
 			for (var j in BattleStatNames) {
 				if (!curSet.evs[j]) continue;
-				if (first) {
+				if (firstEV) {
 					text += 'EVs: ';
-					first = false;
+					firstEV = false;
 				} else {
 					text += ' / ';
 				}
 				text += '' + curSet.evs[j] + ' ' + BattleStatNames[j];
 			}
 		}
-		if (!first) {
+		if (!firstEV) {
 			text += "  \n";
 		}
 		if (curSet.nature) {
 			text += '' + curSet.nature + ' Nature' + "  \n";
 		}
-		var first = true;
-		if (curSet.ivs) {
-			var defaultIvs = true;
-			var hpType = false;
-			for (var j = 0; j < curSet.moves.length; j++) {
-				var move = curSet.moves[j];
-				if (move.substr(0, 13) === 'Hidden Power ' && move.substr(0, 14) !== 'Hidden Power [') {
-					hpType = move.substr(13);
-					if (!exports.BattleTypeChart[hpType].HPivs) {
-						alert("That is not a valid Hidden Power type.");
-						continue;
-					}
-					for (var stat in BattleStatNames) {
-						if ((curSet.ivs[stat] === undefined ? 31 : curSet.ivs[stat]) !== (exports.BattleTypeChart[hpType].HPivs[stat] || 31)) {
-							defaultIvs = false;
-							break;
-						}
-					}
-				}
-			}
-			if (defaultIvs && !hpType) {
-				for (var stat in BattleStatNames) {
-					if (curSet.ivs[stat] !== 31 && curSet.ivs[stat] !== undefined) {
-						defaultIvs = false;
-						break;
-					}
-				}
-			}
-			if (!defaultIvs) {
-				for (var stat in BattleStatNames) {
-					if (typeof curSet.ivs[stat] === 'undefined' || isNaN(curSet.ivs[stat]) || curSet.ivs[stat] == 31) continue;
-					if (first) {
-						text += 'IVs: ';
-						first = false;
+
+		var ivs = curSet.ivs;
+		if (ivs) {
+			text += 'IVs: ';
+
+			var all31 = (ivs.hp === 31 && ivs.atk === 31 && ivs.def === 31 && ivs.spa === 31 && ivs.spd === 31 && ivs.spe === 31);
+			if (all31) {
+				text += '31 all';
+			} else {
+				var firstIV = true;
+				for (var j in BattleStatNames) {
+					var val = curSet.ivs[j];
+					if (val === 31) continue;
+					if (firstIV) {
+						firstIV = false;
 					} else {
 						text += ' / ';
 					}
-					text += '' + curSet.ivs[stat] + ' ' + BattleStatNames[stat];
+					text += '' + val + ' ' + BattleStatNames[j];
 				}
 			}
-		}
-		if (!first) {
-			text += "  \n";
-		}
-		if (curSet.moves) for (var j = 0; j < curSet.moves.length; j++) {
-			var move = curSet.moves[j];
-			if (move.substr(0, 13) === 'Hidden Power ') {
-				move = move.substr(0, 13) + '[' + move.substr(13) + ']';
+			if (!firstIV) {
```

Comment:
This is also no longer necessary, now that `IVs:` is always displayed.

Reply:
On the other hand, `text += '31 all';` can be moved here, so you don't need to manually check for all 31's.

Reply:
In other words, this should be:

```js
if (!firstIV) {
  text += '31 all';
}
text += "  \n";
```

Reply:
We need it to tell whether to add a slash, we don't add a slash before the first IV.

Reply:
Done